android/app/src/main/java/org/electroncash/electroncash3/NewWallet.kt

```diff
@@ -189,34 +193,36 @@ abstract class NewWalletDialog2 : TaskLauncherDialog<String>() {
     override fun doInBackground(): String {
         val name = arguments!!.getString("name")!!
         val password = arguments!!.getString("password")!!
-        onCreateWallet(name, password)
 
-        val multisig = arguments!!.getBoolean("multisig")
+        /**
+         * Handle the resulting keystore.
+         */
+        val ks = onCreateWallet(name, password)
+        if (ks != null) {
+            val masterKey = ks.callAttr("get", "xpub").toString()
+            arguments!!.putString("masterKey", masterKey)
+            arguments!!.putStringArrayList("keystores", updateKeystores(arguments!!, ks))
```

Comment:
This might be difficult to reproduce without adding a delay, but if you modify the arguments in doInBackground, and rotate the screen before doInBackground returns, then onPostExecute may  run on a different Fragment instance and the argument updates will be lost. 

I think the easiest solution is to change the type argument of NewWalletDialog2 so doInBackground returns the keystore to onPostExecute, then move this code to onPostExecute. I don't know why I made it return the name, because that can always be recovered from the dialog arguments.

However, create_multisig should still remain in doInBackground, because it might be slow.

As you're discovering, handling rotation correctly on Android can be a real pain in the ass. I usually test rotating every layout in every state, and I still sometimes miss things.

Reply:
The issue that I had faced here was that the single keystore _had to be_ "processed" in `doInBackground()`, since the lines afterwards check the `keystores` argument to see if it is time to create the multisig wallet.

I hope the solution I came up with is adequate. I also added an additional feature: for multisig wallets, it should now be possible to see master public keys for all cosigners, just like in the desktop app.